hs/src/LedgerState.hs

```diff
@@ -58,8 +59,10 @@ data LedgerValidation = LedgerValidation [ValidationError] LedgerState
 data ValidationError =
   -- | The transaction inputs are not valid.
     BadInputs
-  -- | The transaction results in an increased total balance of the ledger.
-  | IncreasedTotalBalance
+  -- | The transaction fee is too small
+  | FeeTooSmall String
```

Comment:
I would prefer an integer argument here, I understand that just the difference between minimum and actual might not be enought, but why not `FeeTooSmall $MINIMUM $ACTUAL` with two `Natural` parameters?

Reply:
I made it two `Coin`s.